requirements/core.txt

```diff
@@ -4,11 +4,11 @@
 # Without these, iris won't even import.
 
 cartopy
-matplotlib<1.9
-netcdf4
+cf_units
 cftime
+dask>=0.17.1
+matplotlib<1.9
+netcdf4<1.4
```

Comment:
All the dependencies are now in alphabetical order and kept `netcdf4<1.4` to avoid putting the `netcdf4>=1.4` fixes in the same PR.

I have another PR ready once this is merged that should get all the tests, but those depending on `nc-time-axis`, to pass.